src/app/file/pixly_format.cpp

```diff
@@ -0,0 +1,487 @@
+   * optional.  Only call them if you want/need them.  Many of the
+   * transformations only work on specific types of images, and many
+   * are mutually exclusive.
+   */
+
+  /* tell libpng to strip 16 bit/color files down to 8 bits/color */
+  png_set_strip_16(png_ptr);
+
+  /* Extract multiple pixels with bit depths of 1, 2, and 4 from a single
+   * byte into separate bytes (useful for paletted and grayscale images).
+   */
+  png_set_packing(png_ptr);
+
+  /* Turn on interlace handling.  REQUIRED if you are not using
+   * png_read_image().  To see how to handle interlacing passes,
+   * see the png_read_row() method below:
+   */
+  number_passes = png_set_interlace_handling(png_ptr);
+
+  /* Optional call to gamma correct and add the background to the palette
+   * and update info structure.
+   */
+  png_read_update_info(png_ptr, info_ptr);
+
+  /* create the output image */
+  switch (png_get_color_type(png_ptr, info_ptr)) {
+
+    case PNG_COLOR_TYPE_RGB_ALPHA:
+      fop->sequenceSetHasAlpha(true);
+      pixelFormat = IMAGE_RGB;
+      break;
+
+    default:
+      fop->setError("Color type not supported\n)");
+      png_destroy_read_struct(&png_ptr, &info_ptr, NULL);
+      return false;
+  }
+
+  int imageWidth = png_get_image_width(png_ptr, info_ptr);
+  int imageHeight = png_get_image_height(png_ptr, info_ptr);
+
+  // Allocate the memory to hold the image using the fields of info_ptr.
+  rows_pointer = (png_bytepp)png_malloc(png_ptr, sizeof(png_bytep) * height);
+  for (y = 0; y < height; y++)
+    rows_pointer[y] = (png_bytep)png_malloc(png_ptr, png_get_rowbytes(png_ptr, info_ptr));
+
+  for (pass = 0; pass < number_passes; pass++) {
+    for (y = 0; y < height; y++) {
+      png_read_rows(png_ptr, rows_pointer+y, nullptr, 1);
+
+      fop->setProgress(
+        0.5 * ((double)((double)pass + (double)(y+1) / (double)(height))
+        / (double)number_passes));
+
+      if (fop->isStop())
+        break;
+    }
+  }
+
+  {
+    XmlDocumentRef doc = open_xml(fop->filename());
+    TiXmlHandle xml(doc.get());
+    fop->setProgress(0.75);
+
+    TiXmlElement* xmlAnim = xml.FirstChild("PixlyAnimation").ToElement();
+
+    TiXmlElement* xmlInfo = xmlAnim->FirstChild("Info")->ToElement();
+
+    int layerCount  = strtol(xmlInfo->Attribute("layerCount"), NULL, 10);
+    int frameWidth  = strtol(xmlInfo->Attribute("frameWidth"), NULL, 10);
+    int frameHeight = strtol(xmlInfo->Attribute("frameHeight"), NULL, 10);
+
+    UniquePtr<Sprite> sprite(new Sprite(IMAGE_RGB, frameWidth, frameHeight, 0));
+
+    TiXmlElement* xmlFrames = xmlAnim->FirstChild("Frames")->ToElement();
+    int imageCount = strtol(xmlFrames->Attribute("length"), NULL, 10);
+    int frameCount = imageCount / layerCount;
+    sprite->setTotalFrames(frame_t(frameCount));
+
+    for(int i=0; i<layerCount; i++) {
+      sprite->folder()->addLayer(new LayerImage(sprite));
+    }
+
+    int *visible = (int*)malloc(layerCount * sizeof(int));
```

Comment:
`visible` can be a `std::vector`, e.g.

    std::vector<int> visible(layerCount, 0);